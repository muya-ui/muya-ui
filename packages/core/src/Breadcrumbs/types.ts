import React from 'react';

import { ICustomStyleBaseProps, ISizeSpecBaseProps } from '../types';
import { IInlineButtonProps } from '../Button/types';

/**
 *  Breadcrumbs style keys
 * @styles item 每个子项
 * @styles headSeparator 第一个的分割符
 * @styles separator 默认的分隔符
 */
export type IBreadcrumbsStyleKeys = 'item' | 'headSeparator' | 'separator';

export interface IBreadcrumbItem {
  /**
   * 文字内容
   * @default -
   * */
  label: React.ReactNode;
  /**
   * 文字链接
   * @default -
   * */
  url?: string;
  /**
   * 点击事件
   * @default -
   * */
  onClick?: IInlineButtonProps['onClick'];
  /**
   * 鼠标进入事件
   * @default -
   * */
  onMouseEnter?: IInlineButtonProps['onMouseEnter'];
  /**
   * 鼠标离开事件
   * @default -
   * */
  onMouseLeave?: IInlineButtonProps['onMouseLeave'];
}

type IPickFromInlineButtonForBreadcrumbs = Pick<IInlineButtonProps, 'fontWeight'>;
export interface IBreadcrumbsPureProps
  extends ISizeSpecBaseProps,
    IPickFromInlineButtonForBreadcrumbs,
    ICustomStyleBaseProps<IBreadcrumbsStyleKeys> {
  /**
   * 分割符
   * @default ['>']
   **/
  separators?: React.ReactNode[];
  /**
   * 第一个元素单独设置
   * @default -
   * */
  headItem?: IBreadcrumbItem;
  /**
   * 可以传数组
   * @default -
   * */
  items?: IBreadcrumbItem[];
}

export type IBreadcrumbsProps = IBreadcrumbsPureProps & React.HTMLAttributes<HTMLDivElement>;
