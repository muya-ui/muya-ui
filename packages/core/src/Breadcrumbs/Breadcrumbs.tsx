import React, { useCallback, useMemo } from 'react';
import styled, { css } from 'styled-components';

import { GuildRightIcon } from '@muya-ui/theme-light';

import { IButtonType, InlineButton } from '../Button';
import { ICustomStylePropMap } from '../types';
import memoForwardRef from '../utils/memoForwardRef';
import useStyles from '../utils/useStyles';
import useTheme from '../utils/useTheme';
import { withThemeForStyled } from '../utils/withTheme';
import { IBreadcrumbsProps, IBreadcrumbsStyleKeys } from './types';

const separatorWrapperCss = css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
`;
const SeparatorWrapper = styled.span`
  ${separatorWrapperCss}
`;

const SeparatorHeadWrapper = styled.span`
  ${separatorWrapperCss}
`;

const StyledHeadSeparator = styled.span`
  display: inline-flex;
  align-items: center;
`;
const BaseNode = styled.div``;

const BreadcrumbsPure = memoForwardRef<HTMLDivElement, IBreadcrumbsProps>((props, ref) => {
  const {
    size = 'm',
    fontWeight,
    separators,
    children,
    items,
    headItem,
    styles,
    ...otherProps
  } = props;
  const theme = useTheme();
  const token = theme.components.Breadcrumbs;
  const DefaultSeparator = token.separator || GuildRightIcon;
  const childNum = React.Children.count(children);
  const separatorSize = token.separatorSize[size];
  const defaultStyles = useMemo<ICustomStylePropMap<IBreadcrumbsStyleKeys>>(
    () => ({
      item: '',
      headSeparator: '',
      separator: '',
    }),
    [],
  );
  const innerStyles = useStyles('breadcrumbs', defaultStyles, styles);
  const innerSeparators = useMemo(() => {
    if (!separators && headItem) {
      return ['|', '>'];
    }
    return separators || ['>'];
  }, [headItem, separators]);
  const innerItems = useMemo(() => {
    if (headItem && items) {
      items.unshift(headItem);
    }
    return items;
  }, [headItem, items]);

  const renderSeparator = useCallback(
    (key: '>' | '|' | React.ReactNode) => {
      if (key === '|') {
        return (
          <SeparatorHeadWrapper {...innerStyles.headSeparator}>
            <StyledHeadSeparator />
          </SeparatorHeadWrapper>
        );
      } else if (key === '>') {
        return (
          <SeparatorWrapper {...innerStyles.separator}>
            <DefaultSeparator style={{ width: separatorSize, height: separatorSize }} />
          </SeparatorWrapper>
        );
      }
      return key;
    },
    [innerStyles.headSeparator, innerStyles.separator, separatorSize],
  );

  const separatorLastIndex = innerSeparators.length - 1;

  const finalNode = useMemo(() => {
    if (childNum) {
      const parentPropsForChildren = {
        size,
        fontWeight,
      };
      return React.Children.map(children, (child, childIndex) => {
        if (!React.isValidElement(child)) {
          return null;
        }
        const cloneNode = React.cloneElement(child, {
          ...parentPropsForChildren,
          ...child.props,
        });
        let innerNode;
        if (childIndex < childNum - 1) {
          const separatorKey = innerSeparators[childIndex] || innerSeparators[separatorLastIndex];
          innerNode = renderSeparator(separatorKey);
        }

        return (
          <React.Fragment key={childIndex}>
            {cloneNode}
            {innerNode}
          </React.Fragment>
        );
      });
    } else if (innerItems && innerItems.length) {
      return innerItems.map((item, itemIndex) => {
        let innerNode;
        const isLast = itemIndex === innerItems.length - 1;
        const { label: itemLabel, url: itemUrl, ...otherItemProps } = item;
        let href;
        let buttonType: IButtonType = 'normal';
        let constant = isLast;
        if (!isLast) {
          href = itemUrl;
          const separatorKey = innerSeparators[itemIndex] || innerSeparators[separatorLastIndex];
          innerNode = renderSeparator(separatorKey);
          buttonType = 'secondary';
        }
        return (
          <React.Fragment key={itemIndex}>
            <InlineButton
              {...innerStyles.item}
              {...otherItemProps}
              fontWeight="lighter"
              type={buttonType}
              constant={constant}
              size={size}
              href={href}
            >
              {itemLabel}
            </InlineButton>
            {innerNode}
          </React.Fragment>
        );
      });
    }
  }, [
    childNum,
    children,
    fontWeight,
    innerItems,
    innerSeparators,
    innerStyles.item,
    renderSeparator,
    separatorLastIndex,
    size,
  ]);

  return (
    <BaseNode {...otherProps} ref={ref}>
      {finalNode}
    </BaseNode>
  );
});

const BreadcrumbsWithoutDefaultTheme = styled(BreadcrumbsPure)<IBreadcrumbsProps>`
  position: relative;
  display: flex;
  align-items: center;
  ${props => {
    const size = props.size || 'm';
    const token = props.theme.components.Breadcrumbs;
    const separatorSize = token.separatorSize[size];
    return css`
      color: ${token.separatorColor};

      & ${SeparatorHeadWrapper} {
        width: ${token.separatorHeadWidth}px;
      }

      & ${SeparatorWrapper} {
        width: ${token.separatorWidth}px;
      }

      & ${StyledHeadSeparator} {
        border-left: 1px solid ${token.separatorColor};
        height: ${separatorSize}px;
      }
    `;
  }}
`;

const Breadcrumbs = withThemeForStyled(BreadcrumbsWithoutDefaultTheme);

export default Breadcrumbs;
