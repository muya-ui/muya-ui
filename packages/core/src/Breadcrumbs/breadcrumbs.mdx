---
name: Breadcrumbs 面包屑
menu: 组件
route: /components/breadcrumbs
autoDemos: true
parent: 导航
disableDoubleRow: false
---

<DocButtonStyle />

# Breadcrumbs 面包屑

显示当前页面在系统层级结构中的位置，并能快速向上返回。

## 使用场景

- 当系统拥有超过两级及以上的层级结构时；
- 当需要告知用户【我在哪儿】时；
- 当需要向上导航的功能时。

{Demos}

### API

#### IBreadcrumbsProps

<Props propsName="IBreadcrumbsProps" />

#### IBreadcrumbItem

<Props propsName="IBreadcrumbItem" />

### Styles

#### IBreadcrumbsStyleKeys

<Styles typeName="IBreadcrumbsStyleKeys" componentName="breadcrumbs" />
