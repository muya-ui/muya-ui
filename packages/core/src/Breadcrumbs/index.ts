export * from './types';
export { default } from './Breadcrumbs';
export { default as Breadcrumbs } from './Breadcrumbs';
