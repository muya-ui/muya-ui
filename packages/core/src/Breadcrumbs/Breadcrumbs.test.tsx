import React from 'react';
import renderer from 'react-test-renderer';

import Breadcrumbs from './Breadcrumbs';

test('测试 Breadcrumbs 正常情况', () => {
  const tree = renderer
    .create(
      <>
        <Breadcrumbs
          items={[
            { label: 'ss', url: 'ss' },
            { label: 'ss', url: 'ss' },
            { label: 'ss', url: 'ss' },
          ]}
        />
        <Breadcrumbs
          headItem={{ label: 'sdf' }}
          items={[
            { label: 'ss', url: 'ss' },
            { label: 'ss', url: 'ss' },
            { label: 'ss', url: 'ss' },
          ]}
        />
        <Breadcrumbs
          separators={[
            '|',
            '>',
            '|',
            <span
              key="nn"
              style={{
                display: 'inline-flex',
                width: 13,
                justifyContent: 'center',
              }}
            >
              s
            </span>,
          ]}
        >
          <span>sss</span>
          <span>sss</span>
          <span>sss</span>
        </Breadcrumbs>
      </>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
