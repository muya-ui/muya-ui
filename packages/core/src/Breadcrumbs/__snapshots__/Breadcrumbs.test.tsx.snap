// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Breadcrumbs 正常情况 1`] = `
Array [
  .c6 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c7 {
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c3 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(153,153,153,1);
  font-weight: 400;
  font-size: 14px;
  cursor: pointer;
}

.c3:hover {
  color: rgba(51,139,255,1);
}

.c3:active {
  color: rgba(12,108,235,1);
}

.c3 + .c2 {
  margin-left: 8px;
}

.c8 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(102,102,102,1);
  font-weight: 400;
  font-size: 14px;
  cursor: default;
}

.c8 + .c2 {
  margin-left: 8px;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(221,221,221,1);
}

.c0 .c9 {
  width: 25px;
}

.c0 .c4 {
  width: 20px;
}

.c0 .c10 {
  border-left: 1px solid rgba(221,221,221,1);
  height: 8px;
}

<div
    className="c0"
  >
    <a
      aria-disabled={false}
      className="c1 c2 c3 muya-breadcrumbs-item"
      href="ss"
      onClick={[Function]}
      role="button"
    >
      ss
    </a>
    <span
      className="c4 c5 muya-breadcrumbs-separator"
    >
      <svg
        aria-hidden="false"
        className="c6"
        focusable="false"
        style={
          Object {
            "height": 8,
            "width": 8,
          }
        }
        viewBox="0 0 1024 1024"
      >
        <path
          d="M522.187 499.81L262.15 118.426a21.67 21.67 0 015.959-30.121L398.128 3.792a21.67 21.67 0 0129.796 5.851l333.82 490.168a21.67 21.67 0 010 24.378l-333.929 490.276a21.67 21.67 0 01-29.795 5.96l-129.477-84.512a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00-.108-24.703z"
        />
      </svg>
    </span>
    <a
      aria-disabled={false}
      className="c1 c2 c3 muya-breadcrumbs-item"
      href="ss"
      onClick={[Function]}
      role="button"
    >
      ss
    </a>
    <span
      className="c4 c5 muya-breadcrumbs-separator"
    >
      <svg
        aria-hidden="false"
        className="c6"
        focusable="false"
        style={
          Object {
            "height": 8,
            "width": 8,
          }
        }
        viewBox="0 0 1024 1024"
      >
        <path
          d="M522.187 499.81L262.15 118.426a21.67 21.67 0 015.959-30.121L398.128 3.792a21.67 21.67 0 0129.796 5.851l333.82 490.168a21.67 21.67 0 010 24.378l-333.929 490.276a21.67 21.67 0 01-29.795 5.96l-129.477-84.512a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00-.108-24.703z"
        />
      </svg>
    </span>
    <button
      className="c7 c2 c8 muya-breadcrumbs-item"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      ss
    </button>
  </div>,
  .c11 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c1 {
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c8 {
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c3 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(153,153,153,1);
  font-weight: 400;
  font-size: 14px;
  cursor: pointer;
}

.c3:hover {
  color: rgba(51,139,255,1);
}

.c3:active {
  color: rgba(12,108,235,1);
}

.c3 + .c2 {
  margin-left: 8px;
}

.c12 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(102,102,102,1);
  font-weight: 400;
  font-size: 14px;
  cursor: default;
}

.c12 + .c2 {
  margin-left: 8px;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(221,221,221,1);
}

.c0 .c4 {
  width: 25px;
}

.c0 .c9 {
  width: 20px;
}

.c0 .c6 {
  border-left: 1px solid rgba(221,221,221,1);
  height: 8px;
}

<div
    className="c0"
  >
    <button
      className="c1 c2 c3 muya-breadcrumbs-item"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      sdf
    </button>
    <span
      className="c4 c5 muya-breadcrumbs-head-separator"
    >
      <span
        className="c6 c7"
      />
    </span>
    <a
      aria-disabled={false}
      className="c8 c2 c3 muya-breadcrumbs-item"
      href="ss"
      onClick={[Function]}
      role="button"
    >
      ss
    </a>
    <span
      className="c9 c10 muya-breadcrumbs-separator"
    >
      <svg
        aria-hidden="false"
        className="c11"
        focusable="false"
        style={
          Object {
            "height": 8,
            "width": 8,
          }
        }
        viewBox="0 0 1024 1024"
      >
        <path
          d="M522.187 499.81L262.15 118.426a21.67 21.67 0 015.959-30.121L398.128 3.792a21.67 21.67 0 0129.796 5.851l333.82 490.168a21.67 21.67 0 010 24.378l-333.929 490.276a21.67 21.67 0 01-29.795 5.96l-129.477-84.512a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00-.108-24.703z"
        />
      </svg>
    </span>
    <a
      aria-disabled={false}
      className="c8 c2 c3 muya-breadcrumbs-item"
      href="ss"
      onClick={[Function]}
      role="button"
    >
      ss
    </a>
    <span
      className="c9 c10 muya-breadcrumbs-separator"
    >
      <svg
        aria-hidden="false"
        className="c11"
        focusable="false"
        style={
          Object {
            "height": 8,
            "width": 8,
          }
        }
        viewBox="0 0 1024 1024"
      >
        <path
          d="M522.187 499.81L262.15 118.426a21.67 21.67 0 015.959-30.121L398.128 3.792a21.67 21.67 0 0129.796 5.851l333.82 490.168a21.67 21.67 0 010 24.378l-333.929 490.276a21.67 21.67 0 01-29.795 5.96l-129.477-84.512a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00-.108-24.703z"
        />
      </svg>
    </span>
    <button
      className="c1 c2 c12 muya-breadcrumbs-item"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      ss
    </button>
  </div>,
  .c7 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(221,221,221,1);
}

.c0 .c1 {
  width: 25px;
}

.c0 .c5 {
  width: 20px;
}

.c0 .c3 {
  border-left: 1px solid rgba(221,221,221,1);
  height: 8px;
}

<div
    className="c0"
  >
    <span
      size="m"
    >
      sss
    </span>
    <span
      className="c1 c2 muya-breadcrumbs-head-separator"
    >
      <span
        className="c3 c4"
      />
    </span>
    <span
      size="m"
    >
      sss
    </span>
    <span
      className="c5 c6 muya-breadcrumbs-separator"
    >
      <svg
        aria-hidden="false"
        className="c7"
        focusable="false"
        style={
          Object {
            "height": 8,
            "width": 8,
          }
        }
        viewBox="0 0 1024 1024"
      >
        <path
          d="M522.187 499.81L262.15 118.426a21.67 21.67 0 015.959-30.121L398.128 3.792a21.67 21.67 0 0129.796 5.851l333.82 490.168a21.67 21.67 0 010 24.378l-333.929 490.276a21.67 21.67 0 01-29.795 5.96l-129.477-84.512a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00-.108-24.703z"
        />
      </svg>
    </span>
    <span
      size="m"
    >
      sss
    </span>
  </div>,
]
`;
